### PR TITLE
SampleApplication: add sample application for UL3.1 and UL3.2

### PR DESCRIPTION
**Pull request by krystian-hebel**
Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>

### CODE REVIEW

File: Vlv2TbltDevicePkg/SampleApplication/SampleApplication.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
+/** @file
+  Sample UEFI Application 
+**/
```
Comment (pietrushnic) [inline Vlv2TbltDevicePkg/SampleApplication/SampleApplication.c:7]: @krystian-hebel we need correct copyrights.

File: Vlv2TbltDevicePkg/SampleApplication/SampleApplication.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  )
+{
+  EFI_STATUS status = EFI_SUCCESS;
+  EFI_LOADED_IMAGE* loaded_image;
+  EFI_GUID loaded_image_protocol = LOADED_IMAGE_PROTOCOL;
+  EFI_GUID gEfiShellParametersProtocolGuid = EFI_SHELL_PARAMETERS_PROTOCOL_GUID;
+  EFI_SHELL_PARAMETERS_PROTOCOL  *ShellParameters;
+  UINTN i;
+
+  Print(L"Hello World\n");
```
Comment (pietrushnic) [inline Vlv2TbltDevicePkg/SampleApplication/SampleApplication.c]: @krystian-hebel Sample Application?

**Comment by pietrushnic** [comment]
@krystian-hebel I know you follow K&R-like code style, but we should keep edk2 code style - please fix it: https://github.com/tianocore/tianocore.github.io/wiki/Code-Style-C

**Review by pietrushnic — CHANGES_REQUESTED** [review]

**Comment by miczyg1** [comment]
Application works when extracted from image and run by Shell. Also no errors when building.

**Comment by pietrushnic** [comment]
@krystian-hebel can you address my comments?

**Comment by krystian-hebel** [comment]
@pietrushnic I'm currently working on UL4.*, as this isn't in optimum

**Comment by miczyg1** [comment]
@krystian-hebel 3.1 is the part of optimum and a part of this PR

File: Vlv2TbltDevicePkg/SampleApplication/SampleApplication.c (hunk 1 above)
Comment (krystian-hebel) [reply Vlv2TbltDevicePkg/SampleApplication/SampleApplication.c:7]: done

**Review by pietrushnic — APPROVED** [review]

**Comment by krystian-hebel** [comment]
Wait, doesn't compile...

**Comment by krystian-hebel** [comment]
Fixed, works now.

**Comment by miczyg1** [comment]
I would like to create separate branch for verifying requirements regarding sample application. Since it can mess other requirements

**Comment by miczyg1** [comment]
@krystian-hebel this code launches UEFI Shell, but then immediately goes to menu. Is it desired? IIRC 3.2  application should take text as parameter and print it on screen, but I do not see it anywhere

**Comment by krystian-hebel** [comment]
@miczyg1 this code in this form is only capable of building SampleApplication.efi to be extracted and used elsewhere when using `build.sh`. According to @pietrushnic converting it to standalone package would be an overkill, but I think otherwise.

Does it actually start Shell? It shouldn't be able to find out that Shell.efi is internal shell because of different GUID. Rebasing it to `custom_shell_clean` will make it boot, but that's not the point of this branch.

**Comment by miczyg1** [comment]
@krystian-hebel in the PR description You mentioned 3.2 requirement. But it does not verify that requirement. Since we cannot pass any arguments,  they are not printed. To clear things, maybe let's leave 3.1 only in description to avoid confusion. And yes, i got printed UEFI Shell header with version etc.

Is it a big effort to rebase it to `custom_shell_clean`? I am aware, that Shell will be later replaced, but just for clean state, so that noone will complain after flashing the firmware

**Comment by krystian-hebel** [comment]
@miczyg1 we can pass arguments, after extracting/copying `SampleApplication.efi` to an external drive and starting it from there. Rebasing isn't a problem, but it won't change the outcome - application would be in FV, but there is no way of starting it from there.

**Comment by miczyg1** [comment]
@krystian-hebel fine. I did not want to change the outcome. Just leaving clean state is my point. Yes, I tested the SampleApplication after extracting. It works fine.

**Comment by krystian-hebel** [comment]
Rebased and added file for standalone building. Script for this is here: https://gitlab.com/3mdeb/cmr-surgical/uefi-loader/merge_requests/2

**Pull request merged**